### DISCUSSION TITLE
Action functions and CatchBoundaries

### DISCUSSION THREAD

**Discussion by luke-m** [remix-run/remix#1062] [Q&A]
I've been following the Remix Jokes App tutorial and I'm stuck with some very strange behaviour. I've been tinkering around with this, stripped the code of all the unnecessary side noise, gone through a lot of unexplainable moments of "Why does it work now? I didn't change anything 😫" and this is what I've come to.

So my routing structure looks like this:

```
routes
├── jokes
│   ├── index.tsx
│   ├── $jokeId.tsx
│   └── new.tsx
├── index.tsx
├── jokes.tsx
...
```

In `$jokeId.tsx`, there is a `loader` and an `action` function, as well as an `ErrorBoundary` and a `CatchBoundary`. Now, my understanding from the docs and the tutorial is that when you `throw` a `Response` object in either `loader` or `action`, it gets caught in the `CatchBoundary` so you can display an error message in place. This works properly when the `Response` is thrown from `loader`, not however, when it's thrown from `action`. When I try that, I get an error from `jokes.tsx` (the parent route of `$jokeId.tsx`):

```
TypeError: Cannot read property 'user' of undefined
    at `{data.user ? (` (./app\routes\jokes.tsx:54:26)
    at processChild (./node_modules\react-dom\cjs\react-dom-server.node.development.js:3353:14)
...
```

The content of this error isn't really important - what it means is that a `Response` thrown from the `action` function of a child route (`$jokeId.tsx`) seems to trigger a re-render in the parent (`jokes.tsx`) - which I assumed wasn't the case - and that during that re-render, the `loader` function isn't run and thus, the route-component doesn't have data to properly render.

I assume this is a bug, because I've found an odd way to make it work: Adding a `CatchBoundary` to `jokes.tsx` (the parent route) makes the problem go away. It doesn't matter what the new `CatchBoundary` (in `jokes.tsx`) does, it's never executed, because now the other `CatchBoundary` (in `$jokeId.tsx`) is executed and the error message is displayed where I want it. 

I can add a codesandbox if you're having a hard time understanding the problem, but it's a bit complicated to set it up with prisma and everything :D (the CSB)

Best regards

**Comment by ryanflorence** [comment]
I think was fixed in #864, we should have a release out tomorrow, give it a shot then and I think it'll work as expected :)

**Reply by luke-m** [reply]
Thanks, Ryan. I'll check it out and then update this thread. :)

**Reply by luke-m** [reply]
I just updated the `remix` package to 1.1.1, but unfortunately, the problem seems to persist. :/
